fix(ast/estree): fix `JSXOpeningElement` field order

hi-ogawa opened the pull request:
Remaining failure looks like this:

```diff
== tasks/coverage/acorn-test262-diff/acorn-jsx/010.diff ==
@@ -71,7 +71,7 @@
                 "type": "Literal",
                 "start": 19,
                 "end": 26,
-                "value": "&",
+                "value": "&amp;",
                 "raw": "\"&amp;\""
               }
             },
== tasks/coverage/acorn-test262-diff/acorn-jsx/013.diff ==
@@ -30,7 +30,7 @@
                 "type": "Literal",
                 "start": 16,
                 "end": 31,
-                "value": "&&",
+                "value": "&#x0026;&#38;",
                 "raw": "\"&#x0026;&#38;\""
               }
             }
== tasks/coverage/acorn-test262-diff/acorn-jsx/020.diff ==
@@ -90,7 +90,7 @@
                     "type": "JSXText",
                     "start": 31,
                     "end": 53,
-                    "value": "monkeys /> gorillas",
+                    "value": "monkeys /&gt; gorillas",
                     "raw": "monkeys /&gt; gorillas"
                   }
                 ]
== tasks/coverage/acorn-test262-diff/acorn-jsx/028.diff ==
@@ -14,9 +14,7 @@
         "openingFragment": {
           "type": "JSXOpeningFragment",
           "start": 0,
-          "end": 2,
-          "attributes": [],
-          "selfClosing": false
+          "end": 2
         },
         "closingFragment": {
           "type": "JSXClosingFragment",
== tasks/coverage/acorn-test262-diff/acorn-jsx/046.diff ==
@@ -40,7 +40,7 @@
             "type": "JSXText",
             "start": 3,
             "end": 10,
-            "value": "foo>",
+            "value": "foo&gt;",
             "raw": "foo&gt;"
           }
         ]
```

hi-ogawa commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9746?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#9746** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9746?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9746?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#9745** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9745?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9746).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel reviewed (approved):
Great! Thanks loads for doing this. Please go ahead and merge whenever you're ready.

hi-ogawa committed:
fix(ast/estree): fix `JSXOpeningElement` field order (#9746)

Remaining failure looks like this:

```diff
== tasks/coverage/acorn-test262-diff/acorn-jsx/010.diff ==
@@ -71,7 +71,7 @@
                 "type": "Literal",
                 "start": 19,
                 "end": 26,
-                "value": "&",
+                "value": "&amp;",
                 "raw": "\"&amp;\""
               }
             },
== tasks/coverage/acorn-test262-diff/acorn-jsx/013.diff ==
@@ -30,7 +30,7 @@
                 "type": "Literal",
                 "start": 16,
                 "end": 31,
-                "value": "&&",
+                "value": "&#x0026;&#38;",
                 "raw": "\"&#x0026;&#38;\""
               }
             }
== tasks/coverage/acorn-test262-diff/acorn-jsx/020.diff ==
@@ -90,7 +90,7 @@
                     "type": "JSXText",
                     "start": 31,
                     "end": 53,
-                    "value": "monkeys /> gorillas",
+                    "value": "monkeys /&gt; gorillas",
                     "raw": "monkeys /&gt; gorillas"
                   }
                 ]
== tasks/coverage/acorn-test262-diff/acorn-jsx/028.diff ==
@@ -14,9 +14,7 @@
         "openingFragment": {
           "type": "JSXOpeningFragment",
           "start": 0,
-          "end": 2,
-          "attributes": [],
-          "selfClosing": false
+          "end": 2
         },
         "closingFragment": {
           "type": "JSXClosingFragment",
== tasks/coverage/acorn-test262-diff/acorn-jsx/046.diff ==
@@ -40,7 +40,7 @@
             "type": "JSXText",
             "start": 3,
             "end": 10,
-            "value": "foo>",
+            "value": "foo&gt;",
             "raw": "foo&gt;"
           }
         ]
```
